lib/chef_handler_foreman/foreman_facts.rb

```diff
@@ -85,11 +108,26 @@ def get_key(key, prefix)
     end
 
     def send_attributes(attributes)
-      if uploader
-        uploader.foreman_request('/api/hosts/facts', attributes, node.name)
+      if @cachefile and !@@cache_expired
+        Chef::Log.info "No attributes have changed - not uploading to foreman" 
       else
-        Chef::Log.error "No uploader registered for foreman facts, skipping facts upload"
+        if uploader
+          uploader.foreman_request('/api/hosts/facts', attributes, node.name)
+        else
+          Chef::Log.error "No uploader registered for foreman facts, skipping facts upload"
+        end
+      end
+    end
+
+    def verify_checksum(attributes)
+      attrs_checksum = Digest::MD5.hexdigest(attributes.to_s)
+      if File.exist?(@cachefile)
+        contents = File.open(@cachefile, 'r') { |f| f.read }
+        if attrs_checksum.eql?(contents)
+          @@cache_expired = false
```

Comment:
So had to think about it a bit more but there's not a big benefit of reusing it, also some people run the chef-client --once repeatedly from cron. So I'd prefer converting this to just `@cache_expired`, adding related `attr_accessor`. Note you'll have to set this variable to true in the beginning of verify_checksum method. Also initialization on line 24 should be removed.